### ISSUE TITLE
Whether the two quantities govern the estimator pick and ADR-0062's hyperparameters

### ISSUE THREAD

**Issue by DEVunderdog** [DEVunderdog/DataForgeML#526]
## Question

Should the two named quantities govern **which model** the model-based strategies use, and the rest of ADR-0062's data-dependent hyperparameters — and if so, how?

#517 named the quantities (**Usable Rows**, **Rows per Parameter**) and deliberately left the estimator pick alone, tying MICE's floor bound to its cheapest estimator (`BayesianRidge`) so the floor never has to know what the factory will choose. That decoupling is correct for the floor, and it leaves a real gap: **the floor is deliberately permissive for a column that ends up on a hungry estimator.** This ticket closes it.

`_RegressionEstimatorFactory._choose_model` is Doubt 1 in miniature:

```
Linear            -> BayesianRidge
ModerateNonlinear -> RandomForestRegressor
ComplexNonlinear  -> GradientBoostingRegressor  if n_rows >= gradient_boost_min_rows (10_000)
                     RandomForestRegressor      otherwise
```

One nonlinearity tag and one raw row count. Nothing about usable rows, rows per parameter, or the column's own missingness. A `ComplexNonlinear` column with 12,000 rows and 80 predictors gets a gradient booster at 150 rows per parameter, and the config has no way to say otherwise.

To settle:

- **Does the estimator pick get its own floor**, or does it read the same Feasibility Terms the strategy floor already computed? A per-estimator rows-per-parameter bound is the obvious shape, but it would mean the same quantity is compared against two different bounds at two points in `decide`, and the report must not make that look like a contradiction.
- **Which quantity governs the `ComplexNonlinear` GBM-vs-RandomForest pick** — usable rows, rows per parameter, or both — and does a failed bound *downgrade* the estimator or *reject the strategy*? Downgrading keeps the column model-based on a cheaper estimator; rejecting sends it to central tendency. These are materially different outcomes for the same deficit.
- **Do `n_neighbors`, `max_iter`, and `tol` want the quantities at all?** They are dials, not gates. `n_neighbors` is `max(knn_min_neighbors, sqrt(n_features))` clamped by `min(adaptive, n_rows - 1, 25)`, and that `n_rows - 1` term is a raw count where usable rows is arguably meant. `max_iter` and `tol` are shaped by distribution signals and may legitimately want the raw count. The answer may well be **no** for some of these, and saying so explicitly closes the fog patch rather than leaving it half-open.
- **What the estimator choice contributes to the belief record.** #517 fixed the per-candidate record's shape (every term as value / bound / pass-fail, with a feasibility-vs-resource discriminator). If the estimator pick is quantity-driven, its reasoning is a belief too, and #520 needs to know whether it nests inside the winning candidate's record or sits beside it.

Bounds settled here are measured in the same pass as #517's and #518's — this ticket blocks #519 for that reason.

---

*Added by [What composes the weighted signal score](https://github.com/DEVunderdog/DataForgeML/issues/518):*

Two things #518 explicitly handed here rather than settling:

- **`r2_gap` is not a signal-score component.** It answers *which estimator* (linear-ish vs tree-ish), not *whether a model* — so it is this ticket's input, not the score's. It stays fully derivable from the newly-retained `r2_rf` / `r2_linear` pair on `NumericStats`.
- **The capability ladder the three Signal Tiers select from.** #518 settled that the score is a property of the *column*, so it cannot rank MICE against KNN directly; the tiers instead mean "no model / cheapest feasible / richest feasible", and the ordering of the feasible candidates is this ticket's.

Note the mapping is **keyed on `NumericKind`**, not a flat table: the bottom tier's scalar is `Mode` for `BoundedDiscrete` and `Median` elsewhere, because a median is not a valid domain member. That substitution rule is all that remains of the BoundedDiscrete sub-chain's scalar handling.

---

## Amended by [Wayfinder map: Layered imputation door and de-umbrella'd evaluation](https://github.com/DEVunderdog/DataForgeML/issues/528)

Moved to map #528 and reopened. It was closed unresolved when map #516 was abandoned. The maintainer's rule on #528: **no rows-only rule may survive one layer down** — the estimator pick and the dial clamps are covered, whichever layer owns them.

What changes:

- **Drop the belief-record sub-question** ("what the estimator choice contributes to the belief record"). The belief record (#520) is out of scope.
- **Layer placement is not this ticket's question.** [What routing owns and what it returns](https://github.com/DEVunderdog/DataForgeML/issues/529) decides whether the MICE model choice stays in routing; [Where dial defaults and profile estimates are resolved](https://github.com/DEVunderdog/DataForgeML/issues/531) decides where `n_neighbors` / `max_iter` / `tol` are computed. This ticket decides **what quantities they read**, in whichever layer they land.
- **Rigidity check (2) from map #528 applies:** dial defaults must not be frozen into the routing artifact at one `n_rows`.
- Everything else in the question stands: GBM-vs-RF on `gradient_boost_min_rows`, downgrade-vs-reject on a failed bound, the `n_rows - 1` clamp on `n_neighbors`, and `r2_gap` as this ticket's input.

---

*Handed from [What routing owns and what it returns](https://github.com/DEVunderdog/DataForgeML/issues/529) (ADR-0088):*

- The estimator pick **stays in routing** and is **block-level** (`mice_model_choice`). If the quantities govern it, that is routing logic reading the routing profile's row count.
- `n_neighbors`, `max_iter` and `tol` are **not routing**. Whether the quantities feed them is a question for the layer that resolves dials.

---

*Handed from [Where dial defaults and profile estimates are resolved](https://github.com/DEVunderdog/DataForgeML/issues/531) (ADR-0089):*

- Dials resolve in `resolve_recipe(routing, profile, config)`, which takes **no `n_rows` argument**. The `n_rows - 1` clamp on `n_neighbors` reads the recipe profile's row count. It is the only dial operand that reads a row count at all; `max_iter`, `tol` and the rest read profile statistics only. If a quantity replaces it, the recipe layer computes it from the profile.

---

*Handed from [What composes the non-compensatory feasibility floor](https://github.com/DEVunderdog/DataForgeML/issues/517) (ADR-0091):*

- The quantity is **Rows per Predictor** (renamed from Rows per Parameter): Usable Rows ÷ (active numeric − target), capped for MICE by `mice_max_nearest_features`. It is computed in routing, so the estimator pick can read it there.
- The floor returns only failed terms to a private helper; no per-candidate record to nest into.

**Comment by DEVunderdog** [comment]
*Handed from [What composes the weighted signal score](https://github.com/DEVunderdog/DataForgeML/issues/518) (ADR-0092):*

- The middle tier picks the cheapest feasible candidate and the top tier the richest; the ordering is this ticket's capability ladder.
- `RegressionEstimatorFactory` maps `Unpredictable → None`, and the fitter skips the block (`_fitters.py:452`). `Unpredictable` is no longer a router predicate, so the score can send a column with that tag to MICE. Say what it maps to (the recommendation was the cheapest estimator, `BayesianRidge`).

**Comment by DEVunderdog** [comment]
## Resolution

Recorded in **ADR-0094** (`docs/adr/0094-estimator-ladder-steps-down-on-feasibility-terms.md`) and CONTEXT.md (**Capability Ladder**, **Estimator Ladder**, **Estimator Bound**). Amends ADR-0011, 0062, 0089, 0091 and 0092.

- **Capability Ladder: KNN below MICE.** The middle tier takes KNN if feasible, else MICE. The top tier takes MICE if feasible, else KNN. It orders strategies only; tiers never pick an estimator.
- **Estimator Ladder: BayesianRidge < RandomForest < GradientBoosting.** The block's winning `NonlinearityTag` sets the top rung. `r2_gap` enters only through the tag.
- **RF and GBM each get an Estimator Bound** on both Feasibility Terms: `random_forest_min_usable_rows`, `random_forest_min_rows_per_predictor`, `gradient_boost_min_usable_rows`, `gradient_boost_min_rows_per_predictor`. `gradient_boost_min_rows` is deleted. BayesianRidge has no bound; the floor is its bound.
- **A failed bound downgrades one rung. It never rejects.** Rejecting on a block choice would make a column's route depend on its siblings.
- **The block reads its thinnest column.** Each term is the minimum across MICE columns, using the floor's own values (MICE count capped by `mice_max_nearest_features`). Forced columns count.
- **`Unpredictable` → BayesianRidge.** The factory's `None` branch and the fitter's block skip are deleted.
- **Signal on every MICE column**, naming the failed bound, the column and the value, for every skipped rung. No block-level signal slot.
- **Computed in `route` only.** `with_model_choice` and `author` check no bound (informed consent).
- **`n_rows - 1` cap on `n_neighbors` deleted.** sklearn already caps each column at its donor count, which is exactly that column's Usable Rows. The recipe now reads no row count, so rigidity check (2) passes by construction. `k_capped` compares against `knn_max_neighbors` only.
- **`max_iter` and `tol` read no Feasibility Term.** Their operands describe coupling and missingness, not size.

Handed to [Measure the floor bounds, signal weights, and tier boundaries](https://github.com/DEVunderdog/DataForgeML/issues/519): measure the four Estimator Bounds, and check that MICE convergence does not move with Usable Rows.
